esphome/components/debug/debug_component.cpp

```diff
@@ -14,13 +14,26 @@ namespace debug {
 static const char *const TAG = "debug";
 
 void DebugComponent::dump_config() {
+  std::string device_info;
+  device_info.reserve(256);
+
 #ifndef ESPHOME_LOG_HAS_DEBUG
   ESP_LOGE(TAG, "Debug Component requires debug log level!");
   this->status_set_error();
   return;
 #endif
 
+  ESP_LOGCONFIG(TAG, "DebugComponent:");
+  LOG_TEXT_SENSOR("  ", "Device info", this->device_info_);
+  LOG_SENSOR("  ", "Free", this->free_sensor_);
+#ifdef ARDUINO_ARCH_ESP8266
+  LOG_SENSOR("  ", "Fragmentation", this->fragmentation_sensor_);
+  LOG_SENSOR("  ", "Max Block", this->block_sensor_);
```

Comment:
Might be nice to give a bit more information about what these sensors are:
```suggestion
  LOG_SENSOR("  ", "Free space on heap", this->free_sensor_);
#ifdef ARDUINO_ARCH_ESP8266
  LOG_SENSOR("  ", "Heap fragmentation", this->fragmentation_sensor_);
  LOG_SENSOR("  ", "Largest free heap block", this->block_sensor_);
```